tests/backup/backup_delete_test.go

```diff
@@ -670,3 +671,157 @@ var _ = Describe("{DeleteBucketVerifyCloudBackupMissing}", func() {
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			clusterStatus, err = Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			clusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+		})
+
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			var sem = make(chan struct{}, numberOfSimultaneousBackups)
+			var wg sync.WaitGroup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD("Taking %d backups", numberOfBackups)
+			var mutex sync.Mutex
+			for backupLocationUID, backupLocationName := range backupLocationMap {
+				for _, namespace := range bkpNamespaces {
+					for i := 0; i < numberOfBackups; i++ {
+						time.Sleep(timeBetweenConsecutiveBackups)
+						backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+						backupNames = append(backupNames, backupName)
+						sem <- struct{}{}
+						wg.Add(1)
+						go func(backupName, backupLocationName, backupLocationUID, namespace string) {
+							defer GinkgoRecover()
+							defer wg.Done()
+							defer func() { <-sem }()
+							appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+							mutex.Lock()
+							appContextsToBackupMap[backupName] = appContextsToBackup
+							mutex.Unlock()
+							err = CreateBackupWithValidation(ctx, backupName, SourceClusterName, backupLocationName, backupLocationUID, appContextsToBackup, labelSelectors, orgID, clusterUid, "", "", "", "")
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", backupName))
+						}(backupName, backupLocationName, backupLocationUID, namespace)
+					}
+				}
+			}
+			wg.Wait()
+			log.Infof("List of backups - %v", backupNames)
+		})
+
+		Step("Delete all the backups taken in the previous step ", func() {
+			log.InfoD("Deleting the backups")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Unable fetch admin context")
+			backupDriver := Inst().Backup
+			for _, backup := range backupNames {
+				backupUID, err := backupDriver.GetBackupUID(ctx, backup, orgID)
+				dash.VerifySafely(err, nil, fmt.Sprintf("trying to get backup UID for backup %s", backup))
+				_, err = DeleteBackup(backup, backupUID, orgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting the backup %s", backup))
+			}
+			for _, backup := range backupNames {
+				err := DeleteBackupAndWait(backup, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting the backup %s", backup))
```

Comment:
nit: You can say `Waiting for the backup %s to be deleted`

Reply:
Done